docs/api-communication-spec.md

```diff
@@ -0,0 +1,885 @@
+            pcap-api ->> pcap-cli: pcap data
+        end
+    end
+    note over pcap-cli, gorouter: The request was longer than<br/>the gorouter request deadline
+    gorouter --X pcap-cli: Context deadline exceeded
+    gorouter --X pcap-api: Context deadline exceeded
+    par
+        pcap-api ->> pcap-agent1: Stop
+        pcap-api ->> pcap-agent2: Stop
+        pcap-agent1 ->> pcap-api: OK (pcap-agent1)
+        pcap-agent2 ->> pcap-api: OK (pcap-agent2)
+    end
+```
+
+### Invalid Requests
+
+Requests can be invalid for many reasons.
+
+The following diagram presents the following invalid request reasons for BOSH captures:
+
+- Invalid deployment name
+- Invalid instance group
+- No instances matching the request
+
+```mermaid
+sequenceDiagram
+    note over pcap-cli, pcap-api: Scenario: Unknown Deployment
+    pcap-cli ->>+ pcap-api: Token, Capture Request {<br/>BOSH (Deployment, Groups, Instances)<br/>pcap ("eth0", "host 1.2.3.4", 65k) }
+    pcap-api ->> bosh-uaa: Token Keys?
+    bosh-uaa ->> pcap-api: Token Keys
+    note over pcap-api, bosh-uaa: Token and scope verification    
+    pcap-api ->> bosh-director: Instances (Deployment)?
+    bosh-director ->> pcap-api: Unknown deployment: Deployment
+    pcap-api ->> pcap-cli: INVALID_ARGUMENT (Unknown deployment: Deployment)
+    pcap-api -->- pcap-cli: Close
+
+    note over pcap-cli, pcap-api: Scenario: Invalid Instance Group(s) or Instance IDs
+    pcap-cli ->>+ pcap-api: Token, Capture Request {<br/>BOSH (Deployment, Groups, Instances)<br/>pcap ("eth0", "host 1.2.3.4", 65k) }
+    pcap-api ->> bosh-uaa: Token Keys?
+    bosh-uaa ->> pcap-api: Token Keys
+    note over pcap-api, bosh-uaa: Token and scope verification    
+    pcap-api ->> bosh-director: Instances (Deployment)?
+    bosh-director ->> pcap-api: Instances (Deployment)
+    note over pcap-api: List does not contain matching<br/>instance groups(s) or instance IDs
+    pcap-api ->>- pcap-cli: INVALID_ARGUMENT (Requested instances not found)
+```
+
+Availability issues:
+
+- No pcap-agents available in BOSH case (e.g. not deployed for the given deployment, not reachable)
+- `PCAP permission` for the space, under which the application is hosted, is not enabled in the CF case
+
+```mermaid
+sequenceDiagram
+    pcap-cli ->>+ pcap-api: Token, Capture Request {<br/>BOSH (Deployment, Groups, Instances)<br/>pcap ("eth0", "host 1.2.3.4", 65k) }
+    pcap-api ->> bosh-uaa: Token Keys?
+    bosh-uaa ->> pcap-api: Token Keys
+    note over pcap-api, bosh-uaa: Token and scope verification    
+    pcap-api ->> bosh-director: Instances (Deployment)?
+    bosh-director ->> pcap-api: Instances (Deployment)
+    note over pcap-api: Find and check pcap-agent targets
+    note over pcap-api: None found or unable to connect.
+    
+    pcap-api ->>- pcap-cli: FAILED_PRECONDITION (No instances found to capture ...)
+```
+
+The PCAP feature CAN be enabled for a space. If it is not enabled, the capture will fail.
```

Comment:
This should probably be handled separately. If we do not consider this case explicitly the user will only see an error that no instance was found. I'd assume this requires talking the CC, maybe add that to the diagram below

Reply:
I needed to add that to all diagrams that involve the CF case.